File: packages/gbnf-experiment/data/20260908T234247Z_2da5a2c0/ported_implementation/gbnf/grammar_graph/get_input_as_code_points.py
"""Port of ``src/grammar-graph/get-input-as-code-points.ts``."""

from __future__ import annotations

from typing import TYPE_CHECKING, List

if TYPE_CHECKING:  # pragma: no cover
    from .types import ValidInput


def get_input_as_code_points(src: "ValidInput") -> List[int]:
    if not isinstance(src, str):
        return list(src) if isinstance(src, (list, tuple)) else [src]

    return [ord(char) for char in src]

File: proposed-projects/round-trip-experiment/reverse/20260909T015142Z_f8ca89ff/20260910T053604Z_b86749e4/ported_implementation/src/grammar-graph/get-input-as-code-points.ts
import type { ValidInput } from './types';

export const getInputAsCodePoints = (src: ValidInput): number[] => {
  if (typeof src !== 'string') {
    return Array.isArray(src) ? [...src] : [src];
  }

  return src.split('').map((char) => char.charCodeAt(0));
};
